### PR TITLE
fix(core): reject versioning requests using reserved branch names...

### PR DESCRIPTION
**Pull request by cmark**
...or path aliases like HEAD, LATEST and NEXT

### CODE REVIEW

File: commons/com.b2international.index/src/com/b2international/index/revision/DefaultRevisionBranching.java [hunk 1]
```diff
@@ -20,11 +20,14 @@
 import java.util.concurrent.atomic.AtomicLong;
 import java.util.stream.Collectors;
 
+import org.elasticsearch.core.Set;
```
Comment (apeteri) [inline commons/com.b2international.index/src/com/b2international/index/revision/DefaultRevisionBranching.java]:
```suggestion
import java.util.Set;
```

File: commons/com.b2international.index/src/com/b2international/index/revision/RevisionBranch.java [hunk 2]
```diff
@@ -128,20 +123,18 @@
...
+			private final int maximumLength;
+			private final SortedSet<String> reservedBranchNames;
 
-			public BranchNameValidatorImpl(String allowedCharacterSet, int maximumLength) {
+			public Default(String allowedCharacterSet, int maximumLength, Set<String> reservedBranchNames) {
 				this.allowedCharacterSet = allowedCharacterSet;
 				this.maximumLength = maximumLength;
-				pattern = Pattern.compile(String.format("^[%s]{1,%s}(_[0-9]{1,19})?$", allowedCharacterSet, maximumLength));
+				this.reservedBranchNames = reservedBranchNames == null ? ImmutableSortedSet.of() : ImmutableSortedSet.copyOf(reservedBranchNames);
+				this.pattern = Pattern.compile(String.format("^[%s]{1,%s}(_[0-9]{1,19})?$", allowedCharacterSet, maximumLength));
```
Comment (apeteri) [inline commons/com.b2international.index/src/com/b2international/index/revision/RevisionBranch.java]: The 1-19 digit suffix seems to be no longer needed.

**Review by apeteri — APPROVED** [review]
👍

File: commons/com.b2international.index/src/com/b2international/index/revision/RevisionBranch.java (hunk 2 above)
Comment (cmark) [reply commons/com.b2international.index/src/com/b2international/index/revision/RevisionBranch.java]: I'll remove this old, no longer supported feature from the valid branch name pattern.

**Pull request merged**